### ISSUE TITLE
Windows Terminal hangs with DesktopWindowXamlSource

### ISSUE THREAD

**Issue by maksimushka32** [microsoft/terminal#6861]
Windows Terminal doesn't start after reboot. It was working on current machine, but suddenly stop working and now it run only a zero size window with "DesktopWindowXamlSource" caption, that hangs immediately after start. I can close it only from task manager.
![image (2)](https://user-images.githubusercontent.com/54031964/87126584-1ebf7180-c295-11ea-9c50-b0ae6b01b0cb.png)

It was working fine but after another windows update it just doesn't start. I've rebooted my PC several times, deleted profile.json, try to install Preview terminal version - nothing of this doesn't helps to me. 

I'm located in secure local network and don't have administrator rights on my PC.

Link on feddback hub: https://aka.ms/AA8xqz6

# Environment

```
Windows 10 2004
Microsoft Windows NT 10.0.19041.0
Windows Terminal version 1.0.1211.0

```

# Steps to reproduce
Run Windows Terminal from desktop icon.

# Expected behavior
It start and work

# Actual behavior
It start and nothing happen. Just empty window with caption "DesktopWindowXamlSource", that i can't close.

**Comment by DHowett** [comment]
Hey, I'm so sorry we took a long time to get to this bug. Does it still happen with 1.1 or 1.2?

**Comment by maksimushka32** [comment]
Hi!
I've check just now and can tell you that issue still persist. On both 1.1 and 1.2
![image](https://user-images.githubusercontent.com/54031964/90221088-2c2ac700-de12-11ea-9ac2-415c72f05af8.png)

**Comment by zadjii-msft** [comment]
Okay this is a long shot, but I'm not really sure where to go with this one currently. 
* Could you try adding `"showTabsInTitlebar": false,` to your `settings.json`, and see if the Terminal will boot then? This file will be in `%localappdata%\Packages\Microsoft.WindowsTerminalPreview_8wekyb3d8bbwe\LocalState\settings.json`
* Can you open the [PowerToys ](https://github.com/microsoft/PowerToys)Settings app? You can find that by right-clicking on the powertoys icon in the taskbar after installing PowerToys.
![image](https://user-images.githubusercontent.com/18356694/90685836-fc275d80-e22f-11ea-8dc5-508e7e075cd3.png)
	- This one's a real spitball. I know that the PT Settings app is another XAML Islands app, so I'm hoping we can check if the issue is local to the Terminal or not.

**Comment by maksimushka32** [comment]
Hi! 
Sorry for the delay
showTabsInTitlebar setting has no affect on behavior of the terminal. It still hangs.
As for PowerToys unfortunately i can't install it to this PC. I can install apps only from msixbundle file

**Comment by DHowett** [comment]
If you run `wsl.exe` (outside of terminal!) does it ever work, or does it hang forever? How about `wsl.exe -l`?

**Comment by maksimushka32** [comment]
Yes. Problem with Windows Terminal only. Wsl runs nice as from it's own app and from powershell.
I've try to run Terminal even without settings.json and have same result.

**Comment by DHowett** [comment]
Can you use Task Manager to “Create a dump file” for WindowsTerminal.exe? You can do it by right-clicking WindowsTerminal on the Details tab.

Then, upload the “.dmp” file to OneDrive or Google Drive and e-mail me a link. My address is on my GitHub profile.

Thanks!

**Comment by maksimushka32** [comment]
Please, check your e-mail for link

**Comment by DHowett** [comment]
Interesting... it looks like we have a bunch of threads blocked in the _Mcafee security driver_. Are you getting any AV/security alerts from that software?

**Comment by maksimushka32** [comment]
I don't receive any, but maybe security officers does. I will check this at monday.

**Comment by owainkenwayucl** [comment]
I just had this for the first time ever and I don't have McAfee took maybe five minutes and then it closed and resolved itself.

**Comment by owainkenwayucl** [comment]
Oh, more detail I guess.

I just updated to 21H1 - Windows 10.0.19043.1023
Windows Terminal 1.7.1033.0.

I was closing the WIndow after spending a couple of hours logged in via SSH to one of our HPC systems.

**Comment by owainkenwayucl** [comment]
Oh, actually there's an error in event log:

Exception code 0xc0000602

In module KERNELBASE.dll.

**Comment by zadjii-msft** [comment]
Unfortunately, that doesn't give us enough to work with by itself. However, if it's happening consistently, the Feedback Hub might be able to get us some useful info:

/feedback

**Comment by muRn** [comment]
I'm having the same problem as well. Just prior to rebooting Windows Terminal worked fine, but after it just doesn't start. Window titled "DesktopWindowXamlSource" appears for couple seconds and closes itself automatically without any warnings/errors:
![windows_terminal_issue](https://user-images.githubusercontent.com/2263591/119957133-19007680-bfa2-11eb-994d-256389689c97.gif)

I've tried setting `"showTabsInTitlebar": false` in the `settings.json` - didn't help. I've tried running wsl.exe from cmd.exe - works fine. Default profile on Windows Terminal was PowerShell 7 and separately it works fine as well.

Are there any logs written somewhere when Windows Terminal starts? I don't think it's possible to make a memory dump - it's closing itself too fast.

**Comment by astewart-xmos** [comment]
I'm having an identical problem to what @muRn describes.

Additionally, when I try to run it either from cmd.exe or by manually by executing it from its folder in explorer I get this error:

`C:\Users\astewart>wt
The system cannot find the file C:\Users\astewart\AppData\Local\Microsoft\WindowsApps\wt.exe.`

![image](https://user-images.githubusercontent.com/28442350/120328951-5a9e6380-c2b9-11eb-9af5-0f0597e3ffec.png)

**Comment by LinqLover** [comment]
I'm having the same problem right now, this has never happened before on my end. Are there any possibilities to reset some caches or something similar? I don't want to need to use the old conhost again. :D

**Comment by brechtm** [comment]
A reboot (coincidentally during an update to Windows 21H1) fixed this for me.

**Comment by soharang** [comment]
In my case, uninstalling Ubuntu that I installed from Microsoft Store. Both Ubuntu 18.04 LTS and 20.04 LTS causes this problem.
But it does not mean a lot to me if I cannot keep both "Windows terminal" and "Ubuntu WSL".

**Comment by chenfw2017** [comment]
I'm having the same problem right now.   My Environment:
OS:    win10 21H2 19044.1806
Terminal  ver:  1.13.11431.0
:(

**Comment by Agamenox** [comment]
I got this problem when use terminal in administrator mode...

**Comment by mannfuri** [comment]
clear then JSON file, FIX this.
in my case is 
```
C:\Users\user\AppData\Local\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\settings.json
```

**Comment by dariusaurius** [comment]
> clear then JSON file, FIX this. in my case is
> 
> ```
> C:\Users\user\AppData\Local\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\settings.json
> ```

Removing the folder `Microsoft.WindowsTerminal_8wekyb3d8bbwe` fixed it for me

**Comment by Gladrat** [comment]
I've had this issue for a couple of months, and nothing has worked to solve it.
Today I updated my PowerToys app and the DesktopWindowXamlSource window instantly disappeared...

No other manipulation done today.

**Comment by nickgpltech** [comment]
it's one of these settings.
I updated powertoys and same issue persistent.
disabled this, and the ghost window went away.

This seems like more of a powertoys\fancyzones issue than a terminal issue.

![Image](https://github.com/user-attachments/assets/eb6d745e-d41f-465c-9bcc-72b622b2fc1c)

**Comment by majkinetor** [comment]
Yeah, i have this hangin on all the time above WT, and as @nickgpltech told, closing powertoys fixes it.

**Comment by Hassan4Dev** [comment]
Yep, had the same. Specifically, it's the FancyZones tool, and more specifically, it's the "Move newly created windows to their last known zone". Toggling that on / off reliably reproduces this issue.

Edit, you can fix this by adding: DesktopWindowXamlSource under Excluded apps in the FancyZones settings, as such:

<img width="1073" height="181" alt="Image" src="https://github.com/user-attachments/assets/e772ae1d-4332-4d03-b495-7e9d7d124f1a" />

**Comment by Ad-closeNN** [comment]
Hello guys. Today I met this problem.
**TL;DR: Disable `Compatibility > Allow Windows Terminal to run in the background` if you have enable it**
I tried to disable `FancyZones` and uninstall Powertoys but it did't work.
Luckily I found the button then I disabled it. Finally there was no that ghost **DesktopWindowXamlSource**
I don't know if this solution will work for everyone. But if you enable this function I think you should try this. Thank you for watching.

![img](https://geekrewind.com/wp-content/uploads/2025/02/Allow-Window-Terminal-to-run-in-the-background.webp)
Original link of pic: [https://geekrewind.com/wp-content/uploads/2025/02/Allow-Window-Terminal-to-run-in-the-background.webp](https://geekrewind.com/wp-content/uploads/2025/02/Allow-Window-Terminal-to-run-in-the-background.webp)

<details><summary>Original screenshot in Simplified Chinese</summary>
<p>

<img width="1113" height="626" alt="Image" src="https://github.com/user-attachments/assets/88ee3998-24d7-48ac-9f6f-08b74f930949" />

</p>
</details> 

The About Page of Windows Terminal.
Version 1.23.12811.0
<img width="224" height="108" alt="Image" src="https://github.com/user-attachments/assets/73c547c0-8741-4167-96e8-a1d2a9acc4fc" />
